game-app/game-core/src/main/java/games/strategy/engine/framework/ServerGame.java

```diff
@@ -295,9 +300,13 @@ public void startGame() {
       }
       while (!isGameOver) {
         if (delegateExecutionStopped) {
-          // the delegate has told us to stop stepping through game steps
-          // don't let this method return, as this method returning signals that the game is over.
-          Interruptibles.await(delegateExecutionStoppedLatch);
+          if (stopGameOnDelegateExecutionStop) {
+            stopGame();
```

Comment:
I don't quite understand this change?
Is this for testing only? If yes, could this be documented?

Reply:
The test is the only thing that sets this. It's needed so this loop doesn't execute forever doing `Interruptibles.await(delegateExecutionStoppedLatch);`.

I'm not sure how the regular codepath exits this loop - probably by calling stopGame() from a different thread - but that's not something that can be done by the test, since the test body is single-threaded.

Reply:
Added a comment.